pypyr/errors.py

```diff
@@ -16,9 +16,13 @@ class KeyInContextHasNoValueError(ContextError):
     """pypyr context[key] doesn't have a value."""
 
 
-class KeyNotInContextError(ContextError):
+class KeyNotInContextError(KeyError, ContextError):
     """Key not found in the pypyr context."""
 
+    def __str__(self):
+        """KeyError has custom error formatting, avoid this behaviour."""
+        return super(ContextError, self).__str__()
```

Comment:
not clear on what this is doing?

In case of a KeyError, the msg is the missing keyname: KeyError('missing_key_name')

In the case of a ContextError, the msg is whatever str is passed to it: KeyNotInContextError('whatever string here')

When I run the code as is with _or_ without the `__str__` overload, I get the same results.

```context = Context({'a': {'b': 'c'}})
"{a[c]}".format_map(context)
# result <class 'KeyError'> 'c'
print("{x[c]}".format_map(context))
# result: <class '__main__.KeyNotInContextError'> 'x not found in the pypyr context.'
```

Reply:
KeyError does string escaping if the only one parameter passed, so plain string become in double quotes and with quote escaped? 

```
>>> a = KeyNotInContextError("Te'st")
>>> str(a)
'"Te\'st"'
```

Take a look pls at a cpython sources line comment for KeyError

https://github.com/python/cpython/blob/e2af34fcf84b41189b54e1f2912faded5daabaca/Objects/exceptions.c#L1548


Is this expected and I should update 43 failed tests?